### PR TITLE
[wasm][tests] be slightly more careful looking for the connection string

### PR DESCRIPTION
**Pull request by lewing**
Handle wss: and ws: debug proxy connection strings and be slightly more careful about
what we consider a match.

### CODE REVIEW

**Comment by lewing** [comment]
@monojenkins build Linux AArch64

File: sdks/wasm/ProxyDriver/TestHarnessStartup.cs [hunk 1]
```diff
@@ -2,22 +2,24 @@
...
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.FileProviders;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Microsoft.AspNetCore.StaticFiles;
 using Newtonsoft.Json.Linq;
 
 namespace WebAssembly.Net.Debugging {
 	public class TestHarnessStartup {
+		static Regex parseConnection = new Regex (@"listening on (ws?s://.*)");
```
Comment (lewing) [inline sdks/wasm/ProxyDriver/TestHarnessStartup.cs]:
```suggestion
		static Regex parseConnection = new Regex (@"listening on (ws?s://[^\s]*)");
```

File: sdks/wasm/ProxyDriver/TestHarnessStartup.cs [hunk 2]
```diff
@@ -79,10 +81,12 @@ public async Task LaunchAndServe (ProcessStartInfo psi, HttpContext context, Fun
...
 					Console.WriteLine ($"stderr: {str}");
 
-					if (str.Contains ("listening on", StringComparison.Ordinal)) {
-						var res = str.Substring (str.IndexOf ("ws://", StringComparison.Ordinal));
-						if (res != null)
-							tcs.TrySetResult (res);
+					if (tcs.Task.IsCompleted)
+						return;
+
+					var match = parseConnection.Match (str);;
```
Comment (kg) [inline sdks/wasm/ProxyDriver/TestHarnessStartup.cs]: Extra semicolon here

Comment (lewing) [inline sdks/wasm/ProxyDriver/TestHarnessStartup.cs]:
```suggestion
					var match = parseConnection.Match (str);
```

**Comment by lewing** [comment]
@kg we are searching for the string from the chrome* startup output we don't have control over the markers there

**Comment by lewing** [comment]
https://github.com/aslushnikov/getting-started-with-cdp/blob/master/README.md#protocol-fundamentals

**Pull request merged**